### PR TITLE
ASAP-Alchemy: Postera uploading

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR extends the `asap-alchemy predict` CLI to optionally upload predictions to postera manifold. 

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] add uploading to predict
- [x] track the biological target name on alchemy schema
- [x] add tests to ensure column names are correct 

## Questions
- [ ] should we also be tracking the postera mol set name all the way from prep?

## Status
- [x] Ready to go

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/utils.py [hunk 1]
```diff
@@ -25,3 +26,64 @@ def pull_from_postera(molecule_set_name: str):
...
+    postera_uploader = PosteraUploader(
+        settings=PosteraSettings(),
+        molecule_set_name=molecule_set_name,
+        id_field=AlchemyResults.LIGAND_ID.value,
+        smiles_field=AlchemyResults.SMILES.value,
+    )
+
+    _, _, made_new_molset = postera_uploader.push(result_df)
+
+    return made_new_molset
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/cli/utils.py:89]: This doesn't have to return if you don't use the rval.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py [hunk 2]
```diff
@@ -615,8 +648,32 @@ def predict(
     )
     console.print(message)
 
-    # use short switch to find if either value is not None
-    has_ref_data = reference_dataset or experimental_protocol
+    # check if we have a biological target
+    bio_target = target or result_network.target
+
+    # workout if we should upload to postera
+    if bio_target is not None and postera_molset_name is not None:
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py:655]: Warn or log if one if `postera_molset_name` is present without `target`, ie someone probably wanted to upload but forgot to provide target. Currently doesn't give enough info about why no upload if skipped

**Review by hmacdope — CHANGES_REQUESTED** [review]
One nitpick, but other than that looks good! Test it live to make sure all the columns you want are there and then should be good to roll.

**Comment by jthorton** [comment]
okay tested locally and it seems to be working!

**Review by hmacdope — APPROVED** [review]
Legendary

**Pull request merged**